### PR TITLE
NI-DCPower: Enable some LCR compensation metadata

### PR DESCRIPTION
**Pull request by olsl21**
- [x] This contribution adheres to [CONTRIBUTING.md](https://github.com/ni/nimi-python/blob/master/CONTRIBUTING.md).
- [x] I've updated [CHANGELOG.md](https://github.com/ni/nimi-python/blob/master/CHANGELOG.md) if applicable.
- [x] I've added tests applicable for this pull request

### What does this Pull Request accomplish?

- Enable some open/short/load LCR compensation attributes, associated enums and methods
- Add simple system tests to run the perform open/short LCR compensation functions
- Update CHANGELOG.md

### What testing has been done?

The newly added system tests passed.
```
test_perform_lcr_open_compensation[True-None] PASSED
test_perform_lcr_open_compensation[True-additional_frequencies1] PASSED
test_perform_lcr_open_compensation[True-additional_frequencies2] PASSED
test_perform_lcr_short_compensation[True-None] PASSED
test_perform_lcr_short_compensation[True-additional_frequencies1] PASSED
test_perform_lcr_short_compensation[True-additional_frequencies2] PASSED
```

### Additional Info

This PR is part of a series of PRs to enable all the new LCR metadata.

### CODE REVIEW

File: generated/nidcpower/nidcpower/session.py [hunk 1]
```diff
@@ -1018,6 +1018,238 @@ class _SessionBase(object):
...
+    lcr_load_compensation_enabled = _attributes.AttributeViBoolean(1150222)
+    '''Type: bool
+
+    Specifies whether to apply load LCR compensation data to LCR measurements.
+    Both the lcr_open_compensation_enabled and lcr_short_compensation_enabled properties must be set to True in order to set this property to True.
+
+    Use the lcr_open_short_load_compensation_data_source property to define where the load compensation data that is applied to LCR measurements comes from.
+
+    Load compensation data are applied only for those specific frequencies you define with perform_lcr_load_compensation;
+    load compensation is not interpolated from the specific frequencies you define and applied to other frequencies.
```
Comment (sbethur) [inline generated/nidcpower/nidcpower/session.py:1068]: In the C API reference, this is present as a "Note". Why not do the same here?

File: generated/nidcpower/nidcpower/session.py [hunk 2]
```diff
@@ -4756,6 +4988,94 @@ def _parse_channel_count(self):
...
+        Generates open compensation data for LCR measurements based on a default set of test frequencies and, optionally, additional frequencies you can specify.
+
+        You must physically configure an open LCR circuit to use this method to generate valid open compensation data.
+
+        When you call this method:
+
+        -  The open compensation data is written to the onboard storage of the instrument. Onboard storage can contain only the most recent set of data.
+        -  Most NI-DCPower properties in the session are reset to their default values. Rewrite the values of any properties you want to maintain.
+
+        To apply the open compensation data you generate with this method to your LCR measurements, set the lcr_open_compensation_enabled property to **TRUE**.
```
Comment (sbethur) [inline generated/nidcpower/nidcpower/session.py:5004]: Nitpick: The boolean value in Python is `True`, not `TRUE`.

File: generated/nidcpower/nidcpower/session.py [hunk 3]
```diff
@@ -4756,6 +4988,94 @@ def _parse_channel_count(self):
...
+
+        Corrections for frequencies other than the default frequencies or any additional frequencies you specify are interpolated.
+
```
Comment (sbethur) [inline generated/nidcpower/nidcpower/session.py:5007]: I see this snippet of documentation in C function reference missing here: 
image

Why the inconsistency?

File: generated/nidcpower/nidcpower/session.py [hunk 4]
```diff
@@ -4756,6 +4988,94 @@ def _parse_channel_count(self):
...
+        and then call this method on the result.
+
+        Example: :py:meth:`my_session.channels[ ... ].perform_lcr_open_compensation`
+
+        To call the method on all channels, you can call it directly on the :py:class:`nidcpower.Session`.
+
+        Example: :py:meth:`my_session.perform_lcr_open_compensation`
+
+        Args:
+            additional_frequencies (array.array("d")): Defines a further set of frequencies, in addition to the default frequencies, to perform the compensation for. You can specify <=200 additional frequencies.
```
Comment (sbethur) [inline generated/nidcpower/nidcpower/session.py:5023]: We typically let users pass array-like data as `list`, e.g.:
https://github.com/ni/nimi-python/blob/b1bdc6b0a5bed482df775a37968575c39ac05866/generated/nidcpower/nidcpower/session.py#L5453

This enables duck typing and allows users to pass any object that behaves like a list.

File: src/nidcpower/system_tests/test_system_nidcpower.py [hunk 5]
```diff
@@ -724,3 +724,55 @@ def test_wait_for_event_repeated_capabilities(session, channels):
...
+    assert session.lcr_measured_load_resistance == 10.0
+    assert session.lcr_measured_load_reactance == 10.0
+    assert session.lcr_actual_load_resistance == 10.0
+    assert session.lcr_actual_load_reactance == 10.0
+
+
+@pytest.mark.resource_name("4190/0")
+@pytest.mark.options("Simulate=1, DriverSetup=Model:4190; BoardType:PXIe")
+@pytest.mark.parametrize(
+    "compensation_function",
```
Comment (sbethur) [inline src/nidcpower/system_tests/test_system_nidcpower.py:765]:
```suggestion
    "compensation_method",
```

The correct nomenclature for attributes and functions in Python is properties and methods.

File: src/nidcpower/system_tests/test_system_nidcpower.py [hunk 6]
```diff
@@ -724,3 +724,55 @@ def test_wait_for_event_repeated_capabilities(session, channels):
     channels_session = session.channels[channels]
     with channels_session.initiate():
         channels_session.wait_for_event(nidcpower.Event.SOURCE_COMPLETE)
+
+
+@pytest.mark.resource_name("4190/0")
+@pytest.mark.options("Simulate=1, DriverSetup=Model:4190; BoardType:PXIe")
+def test_lcr_attributes(session):
```
Comment (sbethur) [inline src/nidcpower/system_tests/test_system_nidcpower.py:731]: As discussed [here](https://github.com/ni/nimi-python/pull/1740#discussion_r869326979), you can skip get/set tests for properties.

**Review by sbethur — CHANGES_REQUESTED** [review]

File: src/nidcpower/system_tests/test_system_nidcpower.py (hunk 6 above)
Comment (olsl21) [reply src/nidcpower/system_tests/test_system_nidcpower.py]: I've removed these tests in the latest commit

File: src/nidcpower/system_tests/test_system_nidcpower.py (hunk 5 above)
Comment (olsl21) [reply src/nidcpower/system_tests/test_system_nidcpower.py]: Thanks for the info, I've renamed it in the latest commit

File: src/nidcpower/system_tests/test_system_nidcpower.py [hunk 7]
```diff
@@ -724,3 +724,22 @@ def test_wait_for_event_repeated_capabilities(session, channels):
...
+@pytest.mark.options("Simulate=1, DriverSetup=Model:4190; BoardType:PXIe")
+@pytest.mark.parametrize(
+    "compensation_method",
+    [
+        nidcpower.Session.perform_lcr_open_compensation,
+        nidcpower.Session.perform_lcr_short_compensation,
+    ],
+)
+@pytest.mark.parametrize("additional_frequencies", [None, [], [9_000.0, 12_345.0, 12_346.0]])
+def test_perform_lcr_open_short_compensation(
```
Comment (marcoskirsch) [inline src/nidcpower/system_tests/test_system_nidcpower.py:739]: Break this up into two simpler tests, I think that's easier.

**Review by marcoskirsch — CHANGES_REQUESTED** [review]

**Review by marcoskirsch — APPROVED** [review]

File: generated/nidcpower/nidcpower/session.py (hunk 1 above)
Comment (olsl21) [reply generated/nidcpower/nidcpower/session.py:1201]: I've moved this into the "Note" section in my latest commit

File: generated/nidcpower/nidcpower/session.py (hunk 3 above)
Comment (olsl21) [reply generated/nidcpower/nidcpower/session.py:5222]: I've added this snippet for both `perform_lcr_open_compensation()` and `perform_lcr_short_compensation()` in my latest commit

File: generated/nidcpower/nidcpower/session.py (hunk 4 above)
Comment (olsl21) [reply generated/nidcpower/nidcpower/session.py]: I've changed the type of `additional_frequencies` to "list of float" in my latest commit

File: generated/nidcpower/nidcpower/session.py (hunk 2 above)
Comment (olsl21) [reply generated/nidcpower/nidcpower/session.py]: I've updated it to VI_TRUE (which is codegen-ed as "True") in my latest commit

**Review by sbethur — APPROVED** [review]

**Pull request merged**